src/Layout/Header/DeviceSelector/DeviceSelector.js

```diff
@@ -1,7 +1,99 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import PropTypes from 'prop-types';
+/** Actions */
+import { fetchDevicesAction, changeDeviceAction } from '@context/spotify';
 /** Components */
-import Select from '@common/Select';
+import { Popover, PopoverTrigger } from '@common/Popover';
+import { ReactComponent as DevicesIcon } from '@assets/svg/devices.svg';
+import DeviceSelectorItem from './DeviceSelectorItem';
+/** Styled components */
+import {
+  DevicesSelectorWrapper,
+  DevicesSelectorItemsWrapper,
+  NoDevicesFoundText,
+} from './DeviceSelector.styled';
 
-const Avatar = () => <Select />;
+const DeviceSelector = ({ intl }) => {
+  const [forceClose, setForceClose] = useState(false);
 
-export default Avatar;
+  const {
+    spotify: { devices, activeDevice },
+    playing: { current },
+  } = useSelector(state => state);
+
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    activeDevice && dispatch(fetchDevicesAction());
+  }, [activeDevice]);
+
+  const changeDeviceHandler = deviceId => {
+    dispatch(changeDeviceAction(deviceId, current));
+    setForceClose(true);
+  };
+
+  const openSelectorHandler = () => {
+    dispatch(fetchDevicesAction());
+  };
+
+  const closeSelectorHandler = () => {
+    setForceClose(false);
+  };
+
+  const hasActiveDevice = () => {
+    return !!devices.find(device => device.is_active);
```

Comment:
I also have some comments for this line, screw the rest of the files I guess 😆

I'm missing some extra logic here, when I have no device selected and proceed to choose one from the popover the app starts working with it and I can start listening to the music. The problem is that I still see the red icon because we are not updating the devices list until there is a new click in the icon. Maybe we could handle the hasActiveDevice with our redux activeDevice instead?

Reply:
You were right all the way. I don't know how I ended up with that huge mess, but redux activeDevice should do it. Thank you <3